app/src/main/cpp/ControllerDelegate.h

```diff
@@ -70,6 +73,43 @@ class ControllerDelegate {
     std::vector<uint16_t> indices;
   };
 
+  struct HandMeshVertexMSFT {
+    vrb::Vector position;
+    vrb::Vector normal;
+  };
+  struct HandMeshBufferMSFT {
+    std::vector<uint32_t> indices;
+    std::vector<HandMeshVertexMSFT> vertices;
+    uint32_t ibo;
+    uint32_t vbo;
+    HandMeshBufferMSFT() {
+      glGenBuffers(1, &this->ibo);
+      glGenBuffers(1, &this->vbo);
+    }
+    ~HandMeshBufferMSFT() {
+      glDeleteBuffers(1, &this->ibo);
+      glDeleteBuffers(1, &this->vbo);
+    }
+  };
+  typedef std::shared_ptr<HandMeshBufferMSFT> HandMeshBufferPtr;
+  struct HandMeshMSFT {
+    vrb::GeometryDrawablePtr geometry;
+    std::vector<HandMeshBufferPtr> buffers;
+    std::vector<HandMeshBufferPtr> usedBuffers;
+    void Reset() {
+      for(auto buffer: this->buffers)
+        buffer.reset();
+      for(auto buffer: this->usedBuffers)
+        buffer.reset();
+      memset(this, 0, sizeof(HandMeshBufferMSFT));
+    }
+    void Copy(const HandMeshMSFT& aSrc) {
+      this->geometry = std::move(aSrc.geometry);
```

Comment:
hmm I guess this works because calling move on a const reference ends up invoking the copy constructor but it does not seem right to use it like this...

Reply:
Ok, will see how to rework that. The intention here is stealing the pointer, avoiding a deep copy.